File: CS/WindowsApplication1/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraScheduler.UI;
using DevExpress.XtraScheduler;

namespace WindowsApplication1
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void schedulerControl1_EditAppointmentFormShowing(object sender, DevExpress.XtraScheduler.AppointmentFormEventArgs e)
        {
            e.Handled = true;
            MyAppointmentForm f = new MyAppointmentForm(sender as SchedulerControl , e.Appointment, e.OpenRecurrenceForm);
            e.DialogResult = f.ShowDialog();
            
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            schedulerStorage1.Resources.Add(this.schedulerStorage1.CreateResource(1, "John"));
            schedulerStorage1.Resources.Add(this.schedulerStorage1.CreateResource(2, "Jane"));
        }
    }
    public class MyAppointmentForm : AppointmentForm
    {
       public MyAppointmentForm(SchedulerControl sc, Appointment apt, bool openRecurrenceForm)
            : base(sc, apt, openRecurrenceForm)
        {
            this.edtResource.SelectedIndex = -1;
            this.edtResource.Properties.Items.RemoveAt(0);
        }
    }
}

File: VB/WindowsApplication1/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraScheduler.UI
Imports DevExpress.XtraScheduler

Namespace WindowsApplication1
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub schedulerControl1_EditAppointmentFormShowing(ByVal sender As Object, ByVal e As DevExpress.XtraScheduler.AppointmentFormEventArgs) Handles schedulerControl1.EditAppointmentFormShowing
			e.Handled = True
			Dim f As New MyAppointmentForm(TryCast(sender, SchedulerControl), e.Appointment, e.OpenRecurrenceForm)
			e.DialogResult = f.ShowDialog()

		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			schedulerStorage1.Resources.Add(New Resource(1, "John"))
			schedulerStorage1.Resources.Add(New Resource(1, "Jane"))
		End Sub
	End Class
	Public Class MyAppointmentForm
		Inherits AppointmentForm
	   Public Sub New(ByVal sc As SchedulerControl, ByVal apt As Appointment, ByVal openRecurrenceForm As Boolean)
		   MyBase.New(sc, apt, openRecurrenceForm)
			Me.edtResource.SelectedIndex = -1
			Me.edtResource.Properties.Items.RemoveAt(0)
	   End Sub
	End Class
End Namespace